pkg/imgpkg/bundle/images_lock.go

```diff
@@ -26,6 +26,38 @@ type ImagesLock struct {
 	relativeToRepo string
 }
 
+func (o ImagesLock) ImageRefs() []lockconfig.ImageRef {
+	return o.imagesLock.Images
+}
+
+func (o *ImagesLock) Merge(imgLock ImagesLock) error {
+	for _, image := range imgLock.imagesLock.Images {
+		imgRef := image
+		o.imagesLock.AddImageRef(imgRef)
+	}
+
+	return nil
+}
+
+func (o *ImagesLock) GenerateImagesLocations() error {
+	var imgRefs []lockconfig.ImageRef
+	for _, imgRef := range o.imagesLock.Images {
+		imageInBundleRepo, err := o.imageRelativeToBundle(imgRef.Image)
+		if err != nil {
+			return err
+		}
+
+		imgRef.AddLocation(imageInBundleRepo)
+		imgRefs = append(imgRefs, imgRef)
```

Comment:
instead of appending to a new array you can capture loop index and `o.imagesLock.Images[i] = imgRef`